framework/include/utils/FunctionMaterialPropertyDescriptor.h

```diff
@@ -29,18 +29,20 @@ class FunctionMaterialPropertyDescriptor : public DerivativeMaterialPropertyName
    * The descriptor is constructed with an expression that describes the
    * material property.
    * Examples:
-   *   'F'               A material property called 'F' with no declared variable
-   *                     dependencies (i.e. vanishing derivatives)
-   *   'F(c,phi)'        A material property called 'F' with declared dependence
-   *                     on 'c' and 'phi' (uses DerivativeFunctionMaterial rules to
-   *                     look up the derivatives)
-   *   'a:=D[x(t),t,t]'  The second time derivative of the t-dependent material property 'x'
-   *                     which will be referred to as 'a' in the function expression.
+   *   'F'                 A material property called 'F' with no declared variable
+   *                       dependencies (i.e. vanishing derivatives)
+   *   'F(c,phi)'          A material property called 'F' with declared dependence
+   *                       on 'c' and 'phi' (uses DerivativeFunctionMaterial rules to
+   *                       look up the derivatives)
+   *   'a:=D[x(t),t,t]'    The second time derivative of the t-dependent material property 'x'
+   *                       which will be referred to as 'a' in the function expression.
+   *   'x_old:=Old[x]'     The previous time step value of x
+   *   'x_older:=Older[x]' The value of x two time steps ago
    */
```

Comment:
```suggestion
   'F(c,phi, x_old)'          A material property called 'F' with declared dependence
   *                       on 'c', 'phi' and x_old. The DerivativeFunctionMaterial rules are used to compute the derivatives
   *                      with regards to 'c' and 'phi', but 'x_old' is excluded
   */
```
is this true?

Reply:
Uhm, No, I don't think your edit is true. You can still take a derivative w.r.t. an old state. It just doesn't make sense to take a derivative of an old property (at least w.r.t. to current quantities)

Reply:
but would that be the syntax to do it?

Reply:
I guess my questions are:
- how would you mix old and new data (for dependences of x, phi, etc) (if possible)

Reply:
Right now you can get the old state of material properties, but they cannot depend on any variables or extra symbols. Their derivative is always 0.

We may need to re-evaluate this once we add coupling to old/older _variables_. 

What should already work is taking a derivative of an expression w.r.t. the old state of a property.